### PR TITLE
Feature/shield update

### PR DESCRIPTION
**Pull request by vidalmelero**
Implemented feature:

Now the shield bar is attatched to the player. 

Only visible when the player uses the shield.

Shield not visible when player dies

shield visible when recovering shield

### CODE REVIEW

File: Scripts/src/Script/ShieldBar.cpp [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+void ShieldBar::Load(const Config& config)
+{
+	inverted = config.GetBool("Inverted", false);
+	vertical = config.GetBool("Vertical", false);
+	Script::Load(config);
+}
+
+void ShieldBar::SetPlayerPosition(float3 position)
```
Comment (Dartemiss) [inline Scripts/src/Script/ShieldBar.cpp:125]: const reference

File: Scripts/src/Script/UIManager.cpp [hunk 2]
```diff
@@ -128,13 +129,30 @@ void UIManager::UpdateShieldlUI(unsigned int player, float durability) const
...
+	{
+		player1_shield_ui->active = show;
+	}
+	else
+	{
+		player2_shield_ui->active = show;
+	}
+	
+}
+
```
Comment (Dartemiss) [inline Scripts/src/Script/UIManager.cpp:155]: const

**Review by Dartemiss — CHANGES_REQUESTED** [review]
You have to merge develop on this branch.
Also when shield is broken when you press the button nothing has to be shown.
https://gyazo.com/1d467437f6d9aacff1de7066c8687da0

File: Scripts/src/Script/ShieldBar.cpp [hunk 3]
```diff
@@ -122,7 +122,7 @@ void ShieldBar::Load(const Config& config)
 	Script::Load(config);
 }
 
-void ShieldBar::SetPlayerPosition(float3 position)
+void ShieldBar::SetPlayerPosition(const float3 position)
```
Comment (raulgonzalezupc) [inline Scripts/src/Script/ShieldBar.cpp:125]: void ShieldBar::SetPlayerPosition(const float3& position)

File: Scripts/src/Script/ShieldBar.cpp [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+{
+	ShieldBar* instance = new ShieldBar();
+	return instance;
+}
+
+ShieldBar::ShieldBar()
+{
+
+}
+
```
Comment (raulgonzalezupc) [inline Scripts/src/Script/ShieldBar.cpp:28]: with default not need for that

File: Scripts/src/Script/ShieldBar.cpp [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+	vertical = config.GetBool("Vertical", false);
+	Script::Load(config);
+}
+
+void ShieldBar::SetPlayerPosition(const float3 position)
+{
+	player_pos = position;
+}
+
+void ShieldBar::SetShieldStatus()
```
Comment (raulgonzalezupc) [inline Scripts/src/Script/ShieldBar.cpp:130]: this can be const

File: Scripts/src/Script/UIManager.h [hunk 6]
```diff
@@ -9,6 +9,7 @@ class DamageIndicatorSpawner;
 class InputManager;
 class PauseMenuController;
 class ProgressBar;
+class ShieldBar;
 class SpecialProgressBar;
 class ComponentImage;
 class UILives;
```
Comment (raulgonzalezupc) [inline Scripts/src/Script/UIManager.h:15]: line 31 and 32 can be deleted since its comented

Comment (vidalmelero) [reply Scripts/src/Script/UIManager.h:15]: I didn't do that, maybe someone wants to keep that comented

**Comment by raulgonzalezupc** [comment]
image

**Comment by raulgonzalezupc** [comment]
Also, when the shield is breaked, I would suggest to put something on the screen to show the player that he can't use the shield even if he does not press the button. Like showing the bar refilling but again, smaller

**Comment by vidalmelero** [comment]
Okey, now the shield bar is showed too when the shield is recovering, and i set the size a little bit lower.

**Comment by Dartemiss** [comment]
Merge it again with develop for aproval

**Comment by Dartemiss** [comment]
As you created another PR I close this one.

**Pull request closed without merging**